content/webapp/services/prismic/transformers/whats-on.ts

```diff
@@ -0,0 +1,54 @@
+import { clock } from '@weco/common/icons';
+import { isNotUndefined } from '@weco/common/utils/array';
+
+import { isContentList } from '../../../types/body';
+import { FacilityPromo as FacilityPromoType } from '../../../types/facility-promo';
+import { Page as PageType } from '../../../types/pages';
+
+/** The What's On page in Prismic includes a content list which is used to pick
+ * items for the 'Try these too' promo section.
+ *
+ * This extracts them from the page, and also adds a bit of extra info we can't
+ * easily record in Prismic.
+ *
+ */
+
+export function getTryTheseTooPromos(
+  whatsOnPage: PageType
+): FacilityPromoType[] {
+  return whatsOnPage.body
+    .filter(isContentList)
+    .filter(slice => slice.value.title === 'Try these too')
+    .flatMap(slice => slice.value.items)
+    .map(item =>
+      item.type === 'pages'
+        ? {
+            id: item.id,
+            url: `/pages/${item.id}`,
+            title: item.title,
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            image: item.promo!.image!,
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            description: item.promo!.caption!,
+          }
+        : undefined
+    )
+    .filter(isNotUndefined);
+}
+
+export function enrichTryTheseTooPromos(
+  promos: FacilityPromoType[]
+): FacilityPromoType[] {
+  const facilityPromoMetas = {
+    'The Reading Room': {
+      metaIcon: clock,
+      metaText: 'Open during gallery hours',
+    },
+  };
+
+  return promos.map(p => ({
+    ...p,
+    // Add any extra meta information which can't be recorded in Prismic
+    ...(facilityPromoMetas[p.title] ? facilityPromoMetas[p.title] : {}),
```

Comment:
Could we use the facilityPromo id to determine which one it is, in case the title changes for some reason? We have other[ prismic ids in the codebase](https://github.com/wellcomecollection/wellcomecollection.org/blob/0222ac12c268e55034f15a0b0dcc406d5ea779d9/common/data/hardcoded-ids.ts#L6) for similar purposes.

Reply:
Good suggestion, thanks!